test/unity/toolkits/drawing_classifier/test_dc_serialization.cxx

```diff
@@ -125,6 +116,15 @@ BOOST_AUTO_TEST_CASE(test_dc_init_model) {
     TS_ASSERT_EQUALS(dense_layer.innerproduct().outputchannels(), 128);
   }
 
+  layer_index++;
+  {
+      const auto& relu_layer = nn.layers(layer_index);
```

Comment:
Does using `auto` make the code more or less readable?

Reply:
The type is generated by protobuf: 
```
::CoreML::Specification::NeuralNetworkLayer&
```
`auto` can better decouple the underlying protobuf structure from the application layer if we have any change in protobuf message protocol.

Furthermore, the fully qualified name of this type is hard to memorize.